### DISCUSSION TITLE
Display problems

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#51] [Q&A]
@marcel-licence I have firstly connected an 1.8" 120×160 display to the ESP32 Audio-kit and have to add the initialization line code which I have got from the Adafruit library example and also uncommented the line for st7735 compatible display in 'config.h', and the display works fine.
But now I bought a 80×160 st7735 display and it didn't work, agian I had to edit the initialization line code that you written in the code and had to uncomment the line for st7735 compatible display in config.h. But the display shows like this 👇 it is inverted and the colours are different.

![IMG20210829154215](https://user-images.githubusercontent.com/53053463/131252192-76b232ce-f852-4089-a699-398e5b640a60.jpg)

These are the lines I had to edit:-
![IMG-20210829-WA0074](https://user-images.githubusercontent.com/53053463/131252530-13a0ce76-a001-478d-8347-797037fdbafa.jpg)
![IMG-20210829-WA0073](https://user-images.githubusercontent.com/53053463/131252536-1ae7d08c-3140-4905-bd78-20aaf2c20156.jpg)

**Comment by marcel-licence** [comment]
I am using a modified version of the library: https://github.com/marcel-licence/Adafruit-ST7735-Library
Please try this

**Reply by souravlal** [reply]
But even in this library in examples the initialisation line of code for 0.96" display is different than the one which you have used in your program.

**Reply by souravlal** [reply]
I have tried it with your modified library the texts are mirrored, it is the problem of my display. I have tried it with examples in the adafruit library that also displays mirror images. Is there any way to mirror the display text in the code so that I can actually get the correct orientation.

![IMG20210830114638](https://user-images.githubusercontent.com/53053463/131312376-8493ba77-8959-4b88-9b25-9765d7914cba.jpg)

Is it necessary to uncomment the line to activate the st7735 compatible display in 'config.h'?

But having said that I still can't load a new sample from SD Card, still needs a reboot to load new samples. Otherwise previously selected sample gets triggered. I am using MIDI Channel 2.
My settings looks like this:-

![IMG20210830132237](https://user-images.githubusercontent.com/53053463/131312553-72c47598-e732-422d-a429-ed5e67e12f0d.jpg)

**Reply by marcel-licence** [reply]
The original library does not support this display. It allows displaying something but with some errors on screen. 
Did you ensure that you have the correct initR call: 'tft.initR(INITR_096_160x80_IPS);  // Init ST7735S mini display' ?
And yes 'DISPLAY_160x80_ENABLED' shall be active because that is the screen you are using.

**Reply by souravlal** [reply]
The Original Adafruit library do support this display, I have tried their graphicstest.ino sketch and they call this display 'mini 80×160'.
But the only problem is the texts are mirrored.

**Reply by marcel-licence** [reply]
In my version there were was an incorrect offset used to display the data. There was also an issue reported to adafruit: https://github.com/adafruit/Adafruit-ST7735-Library/issues using a specific version of the 160x80 display.

**Reply by Sicco55** [reply]
I had the same problem with the cheap clones of the ST7735S display with the blue PCB. Solve the mirror problem: change in Adafruit_ST7xx.h -> #define ST77XX_MADCTL_MX 0x36 // It was 0x40. This solved the problem! I use: tft.initR(INITR_MINI160x80);

**Reply by Sicco55** [reply]
Correction! 
Change in Adafruit_ST735.cpp :

else if (options == INITR_096_160x80_IPS) {
#define ST77XX_MADCTL_MX 0x36 // Changed 36 for blue PCB
        _height = ST7735_TFTWIDTH_80;
        _width = ST7735_TFTHEIGHT_160;
        displayInit(Rcmd2green160x80);
        _colstart = 24; //was 26 
        _rowstart = 0;  //was 1

**Reply by marcel-licence** [reply]
Did you had also the same problems by using INITR_096_160x80_IPS and also INITR_MINI160x80? 
I've added INITR_096_160x80_IPS  a view month ago into my fork because INITR_MINI160x80 didn't work for me.
Otherwise it looks like we need some more definitions for the IPS 160x80 displays if there are more incompatible models on the marked.

**Reply by Sicco55** [reply]
Yes, both init rules did had the same effect. I tried a lot different libraries and settings but only this modification helped. These cheap (chinese Ali) clones are not 100% compatible. The black board ST7735S is even worse. It only showed only 120x80. That problem is solved a few weeks ago by Boyeen in the TFT_eSPI library. Although it is sold as a ST7735 it is actually a GC9106. The black board is not usable for this project. More problems with this diplays can be expected.

**Reply by marcel-licence** [reply]
Hm, very confusing. I guess that adafruit will focus more on their own products. I had also issues in the past but I tried to change the SPI clock speed. Thanks a lot for sharing your discovery.
